packages/expo/src/dom/www-types.ts

```diff
@@ -10,3 +12,40 @@ export type BridgeMessage<TData extends JSONValue> = {
   type: string;
   data: TData;
 };
+
+export type MarshalPropValuePrimitiveType = number | string | boolean | null | undefined;
+export type MarshalPropValueNativeAction = (...args: any[]) => any;
+export type MarshalPropValueType =
+  | MarshalPropValuePrimitiveType
+  | MarshalPropValuePrimitiveType[]
+  | Record<string, MarshalPropValuePrimitiveType>
+  | MarshalPropValueNativeAction;
+
+export { WebViewProps };
+
+/**
+ * The explicit props for DOM components.
+ */
+interface DOMComponentProps {
+  /**
+   * The props passing to WebView.
+   */
+  dom?: Omit<WebViewProps, 'source'>;
+
+  /**
+   * Whether to resize the native WebView size based on the DOM content size.
+   * @default false
+   */
+  autoSize?: boolean;
```

Comment:
Probably better to add this to the `dom` object or prefix it with `dom` so we aren't squatting too many props. The number of hidden API values that users should have to learn needs to be minimal.